stl/inc/ranges

```diff
@@ -534,6 +535,332 @@ namespace ranges {
+
+        constexpr explicit iota_view(_Wi _Value_) noexcept(
+            is_nothrow_move_constructible_v<_Wi>&& is_nothrow_default_constructible_v<_Bo>) // strengthened
+            : _Value(_STD move(_Value_)) {}
+
+        constexpr iota_view(type_identity_t<_Wi> _Value_, type_identity_t<_Bo> _Bound_) noexcept(
+            is_nothrow_move_constructible_v<_Wi>&& is_nothrow_move_constructible_v<_Bo>) // strengthened
+            : _Value(_STD move(_Value_)), _Bound(_STD move(_Bound_)) {
+            if constexpr (totally_ordered_with<_Wi, _Bo>) {
+                _STL_ASSERT(_Value_ <= _Bound_, "Per N4878 [range.iota.view]/8, the first argument must precede the "
+                                                "second when their types are totally ordered.");
+            }
+        }
+
+        constexpr iota_view(_It _First, _Se _Last) noexcept( // Per LWG-3523
+            is_nothrow_move_constructible_v<_Wi>&& is_nothrow_move_constructible_v<_Bo>) // strengthened
+            : _Value(_STD move(_First._Current)), _Bound(_STD move(_Bound_from(_Last))) {}
+
+        _NODISCARD constexpr _It begin() const noexcept(is_nothrow_copy_constructible_v<_Wi>) /* strengthened */ {
+            return _It{_Value};
+        }
+
+        _NODISCARD constexpr _Se end() const noexcept(is_nothrow_copy_constructible_v<_Bo>) /* strengthened */ {
+            if constexpr (same_as<_Bo, unreachable_sentinel_t>) {
+                return unreachable_sentinel;
+            } else {
+                return _Se{_Bound};
+            }
+        }
+
+        // clang-format off
+        _NODISCARD constexpr auto size() const noexcept(noexcept(_Bound - _Value)) /* strengthened */
+            requires (same_as<_Wi, _Bo> && _Advanceable<_Wi>)
+                || (integral<_Wi> && integral<_Bo>)
+                || sized_sentinel_for<_Bo, _Wi> {
+            // clang-format on
+            if constexpr (_Integer_like<_Wi> && _Integer_like<_Bo>) {
+#pragma warning(suppress : 4146) // unary minus operator applied to unsigned type, result still unsigned
+                return (_Value < 0) ? ((_Bound < 0) ? (_To_unsigned_like(-_Value) - _To_unsigned_like(-_Bound))
+#pragma warning(suppress : 4146) // unary minus operator applied to unsigned type, result still unsigned
+                                                    : (_To_unsigned_like(_Bound) + _To_unsigned_like(-_Value)))
+                                    : (_To_unsigned_like(_Bound) - _To_unsigned_like(_Value));
+            } else {
+                return _To_unsigned_like(_Bound - _Value);
+            }
+        }
+    };
+
+    // clang-format off
+    template <class _Wi, class _Bo>
+        requires (!_Integer_like<_Wi> || !_Integer_like<_Bo>
+            || (_Signed_integer_like<_Wi> == _Signed_integer_like<_Bo>))
+    iota_view(_Wi, _Bo) -> iota_view<_Wi, _Bo>;
+    // clang-format on
+
+    template <class _Wi, class _Bo>
+    inline constexpr bool enable_borrowed_range<iota_view<_Wi, _Bo>> = true;

```

Comment:
I observe that WG21-N4878 [ranges.syn] depicts `template<weakly_incrementable W, semiregular Bound>` here. I don't think that the constraints matter, but wanted to double-check.

Reply:
if this specialization is triggered for `iota_view<T, U>`, we know that someone somewhere formed the _template-id_ `iota_view<T, U>. When they did, the compiler checked that `T` and `U` satisfy `iota_view`'s constraints ([[temp.names]/7](http://eel.is/c++draft/temp.names#7)). Consequently it's redundant to check those constraints again for cases like this where we're matching a specialization, the only effect is to increase the number of places we have to keep the set of constraints synchronized.

I'll see if I can convince the editors that removing constraints here and on `enable_borrowed_range<subrange<I, S, K>>` is editorial (cplusplus/draft#4519).